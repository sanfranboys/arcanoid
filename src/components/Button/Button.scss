.button-component {
  width: 100%;
  padding: 12px 20px;
  color: var(--white);
  background-color: transparent;
  border: 2px solid var(--color-border-dark);
  cursor: pointer;
}

.button-component:hover {
  background-color: var(--mainHover);
}
