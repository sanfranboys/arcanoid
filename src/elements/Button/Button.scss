.button-component {
  width: 100%;
  padding: 12px 20px;
  color: var(--white);
  background-color: transparent;
  border: 2px solid var(--color-border-dark);
  cursor: pointer;

  &:hover {
    background-color: var(--mainHover);
  }

  &__disabled {
    opacity: 0.5;

    &:hover {
      background-color: inherit;
    }
  }
}
