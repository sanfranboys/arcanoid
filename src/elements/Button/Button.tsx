import React, { ButtonHTMLAttributes, FC } from 'react'
import classNames from '@/utils/'
import './Button.scss'

const Button: FC<ButtonHTMLAttributes<HTMLButtonElement>> = ({
  children,
  onClick,
  className,
  type = 'button',
  disabled,
}) => (
  <button
    className={classNames(className, 'button-component', {
      'button-component__disabled': disabled,
    })}
    type={type}
    onClick={onClick}
    disabled={disabled}
  >
    {children}
  </button>
)

export default Button
