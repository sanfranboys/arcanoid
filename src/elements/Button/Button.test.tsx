import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow, ShallowWrapper } from 'enzyme'
import Button from './Button'

describe('Button testing', () => {
  let wrapper: ShallowWrapper
  const mockClick = jest.fn()
  const props = {
    className: 'test-class',
    children: 'Push me',
    onClick: mockClick,
  }
  beforeEach(() => {
    wrapper = shallow(<Button {...props} />)
  })

  it('Button renders correctly', () => {
    const tree = create(<Button {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('component event testing', () => {
    it('Button handleClick works', () => {
      wrapper.simulate('click')
      expect(mockClick.mock.calls.length).toEqual(1)
    })
  })
})
