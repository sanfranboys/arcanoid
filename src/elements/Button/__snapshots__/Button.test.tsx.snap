// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing Button renders correctly 1`] = `
<button
  className="test-class button-component"
  onClick={[MockFunction]}
  type="button"
>
  Push me
</button>
`;
